app/controllers/declaration/ItemsSummaryController.scala

```diff
@@ -63,11 +63,22 @@ class ItemsSummaryController @Inject()(
       .map(_ => Redirect(controllers.declaration.routes.ProcedureCodesPageController.displayPage(newItem.id)))
   }
 
-  /*
-  Compilation warning
-    match may not be exhaustive.
-    [warn] It would fail on the following input: Left(_)
-   */
+  def removeItem(itemId: String): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
+    exportsCacheService.get(journeySessionId) flatMap {
```

Comment:
could we use the get item by id and sessionid here so we can do everything in one go?

Reply:
Afraid not, that method just  returns the Export Item when I  need the whole cache object to be able to do the update

Reply:
This kind of logic could potentially be pulled down into the service layer though I guess

Reply:
yeah maybe. I suppose if you get the item.... then get the cachemodel to update its and extra db call. Your approach avoids that.